donate/payments/views.py

```diff
@@ -326,8 +325,10 @@ def form_valid(self, form, send_data_to_basket=True):
 
             # Create a subcription against the payment method
             result = gateway.subscription.create({
-                'plan_id': self.get_plan_id(self.currency),
-                'merchant_account_id': self.get_merchant_account_id(self.currency),
+                'plan_id': get_plan_id(self.currency),
+                'merchant_account_id': get_merchant_account_id_for_paypal(
+                    self.currency, form.cleaned_data['amount']
+                ),
```

Comment:
Ok, found the problem: we need to pass in the `'first_billing_date'` and that gets it to work for micropayments. Not sure why it already works for macro payments since it goes through the same logic though.